ultimatepython/data_structures/heap.py

```diff
@@ -0,0 +1,66 @@
+"""
+Heaps are a special type of binary tree-based data structure that
+maintains a specific order among the elements. This tutorial covers
+how to create, manipulate, and use heaps in Python.
+"""
+
+import heapq
+
+
+def main():
+    # Creating a min-heap
+    min_heap = []
+    heapq.heappush(min_heap, 3)
+    heapq.heappush(min_heap, 1)
+    heapq.heappush(min_heap, 4)
+    heapq.heappush(min_heap, 1)
+    heapq.heappush(min_heap, 5)
+    assert min_heap[0] == 1  # The root of the heap is the smallest element
+
+    # Pop the smallest element
+    smallest = heapq.heappop(min_heap)
+    assert smallest == 1
+
+    # Creating a max-heap
+    max_heap = []
+    nums = [3, 1, 4, 1, 5]

```

Comment:
Can we define this array at the top of the `main` function so that it can be reused for the min-heap as well?